example/src/main/java/com/stripe/example/activity/PaymentActivity.java

```diff
@@ -173,4 +181,66 @@ private void initListView() {
                 new int[]{R.id.last4, R.id.tokenId});
         listView.setAdapter(simpleAdapter);
     }
+
+    private void saveCreditCard() {
+        Card cardToSave = createCardToSave();
+
+        if (validateCard(cardToSave)) {
+            startProgress();
+            new Stripe().createToken(
+                    cardToSave,
+                    PUBLISHABLE_KEY,
+                    new TokenCallback() {
+                        public void onSuccess(Token token) {
+                            addToList(token);
+                            finishProgress();
+                        }
+                        public void onError(Exception error) {
+                            handleError(error.getLocalizedMessage());
+                            finishProgress();
+                        }
+                    });
+        }
+    }
+
+    private void saveCreditCardWithRx() {
+        final Card cardToSave = createCardToSave();
+        if (!validateCard(cardToSave)) {
+            return;
+        }
+
+        final Stripe stripe = new Stripe();
+        final Observable<Token> tokenObservable =
+                Observable.fromCallable(
+                        new Callable<Token>() {
+                            @Override
+                            public Token call() throws Exception {
+                                startProgress();
```

Comment:
It's observed on the UI thread anyhow, so  calling `finishProgress` there shouldn't be an issue